components/DatePicker/DatePicker.jsx

```diff
@@ -47,37 +47,46 @@ export default class DatePicker extends React.Component<Props> {
   }
 
   state = {
-    open: false,
     valid: true,
   }
 
   componentWillMount() {
-    const defaultValue = moment(this.props.defaultValue || "")
-    if(defaultValue.isValid()) this.handleOnChange(defaultValue)
+    this.props.onChange(this.defaultValue())
+  }
+
+  fieldFormat = () => {
+    const { timeFormat, dateFormat } = this.props
+    return timeFormat && timeFormat !== "" ? "YYYY-MM-DDTHH:mm:00Z" : dateFormat
   }
 
   handleOnChange = (dateTime) => {
-    const {timeFormat} = this.props
-    if(dateTime.format && dateTime.isValid()) {
-      this.setState({valid: true})
-      const dt = this.props.timeZone ? moment.tz(dateTime) : dateTime
-      setTimeout(() => {
```

Comment:
I'm not really sure why there's a `setTimeout` in here, at a first glance it seems unnecessary, but @thatjsguy would know better. If that's somehow a workaround the fact the `setState` call is async, we should instead be using the `setState` onComplete callback, as in:

```jsx
const dt = this.props.timeZone ? moment.tz(dateTime) : dateTime
const onComplete = () => {
  this.props.onChange(dt.format(timeFormat && timeFormat !== "" ? "YYYY-MM-DDTHH:mm:00Z" : "YYYY-MM-DD"))
}

this.setState({ valid: true }, onComplete)
```

Reply:
That's removed code...

Reply:
@drborges exactly what that was originally for - this is better 👍